test/datasets/test_iwslt2016.py

```diff
@@ -150,7 +152,7 @@ def tearDownClass(cls):
     ])
     def test_iwslt2016(self, split, src, tgt, dev_set, test_set):
 
-        root_dir = self.get_base_temp_dir()
+        root_dir = tempfile.TemporaryDirectory().name
```

Comment:
qq, why is this used in `test_iwslt2016`, but not in `test_iwslt2016_split_argument`?

Reply:
It's a good catch. I should use unique base directories in `test_iwslt2016_split_argument ` as well, let me push this change. 

To answer why we need unique base directories with every test: It seems keeping same base directory lead to tests failure that I didn't completely get grasp of. My suspicious is since we are writing and extracting partial (we only write files that are requested in test) dataset  in the same base directory with every parameterized test, this is leading to some weird conflicts in caching and extraction of dataset. So to avoid this situation, it's best to generate new base directory for every parameterized test. Although, I also do see need for better explanation of what's really going on under the hood.

Reply:
Can we also use a context manager here to ensure that the temp directory get's cleaned up? Usually this would be handled by the `TempDirMixin` class but this test is a special case.

Reply:
good catch, let me do that!